sycl/include/sycl/builtins.hpp

```diff
@@ -1082,12 +1084,14 @@ detail::enable_if_t<detail::is_ugeninteger<T>::value, T> abs(T x) __NOEXC {
   return __sycl_std::__invoke_u_abs<T>(x);
 }
 
-// ugeninteger abs (geninteger x)
+// igeninteger abs (geninteger x)
 template <typename T>
-detail::enable_if_t<detail::is_igeninteger<T>::value,
-                    detail::make_unsigned_t<T>>
-abs(T x) __NOEXC {
-  return __sycl_std::__invoke_s_abs<detail::make_unsigned_t<T>>(x);
+detail::enable_if_t<detail::is_igeninteger<T>::value, T> abs(T x) __NOEXC {
+  auto res = __sycl_std::__invoke_s_abs<detail::make_unsigned_t<T>>(x);
```

Comment:
I tried and faced with some incompatibility with HIP and libclc, so instead changed the SYCL high-level API. Changing SYCL API (and not OpenCL's __invoke***) also looks cleaner for me, because there is no OpenCL's abs builtin which returns signed type.